server/auth/gatekeeper.go

```diff
@@ -370,3 +371,10 @@ func DefaultClientForAuthorization(authorization string) (*rest.Config, *servert
 		Kubernetes:  kubeClient,
 	}, nil
 }
+
+func mergeServerRestConfig(argoServerConfig *rest.Config, newConfig *rest.Config) *rest.Config {
```

Comment:
oh I meant using a struct merge (which may require a library), but this is fine too